tests/zfs-tests/include/math.shlib

```diff
@@ -42,6 +42,30 @@ function within_percent
 	return 1
 }
 
+#
+# Return 0 if value is within +/-tolerance of target.
+# Return 1 if value exceeds our tolerance.
+# For use like this:
+#
+# Do $action if value is within the tolerance from target passed in:
+#	within_tolerance VAL TAR TOL && $action
+# Do $action if value surpasses the tolerance from target passed in:
+#	within_tolerance VAL TAR TOL || $action
+#
+function within_tolerance #value #target #tolerance
+{
+	typeset val=$1
+	typeset target=$2
+	typeset tol=$3
+
+	typeset diff=$(echo "scale=2; $val - $target" | bc)
+	diff=$(echo "scale=2; if ($diff < 0) ($diff * -1.0) else $diff" | bc)
```

Comment:
[nit] I'd suggest adding an `abs(x)` function to our `math.shlib` library to do this since it would be useful.  I thought ksh might already provide this functionality, but it doesn't appear too.

I'm assuming you opted for this appraoch since `within_percent` does the same thing to handle floating point values?  That makes sense when working with percentages, but it seems a little less useful in this context.